src/main/java/org/jitsi/jigasi/transcription/OracleTranscriptionService.java

```diff
@@ -0,0 +1,350 @@
+
+    private static final String configFilePath = JigasiBundleActivator.getConfigurationService()
+            .getString(OCI_AUTH_CONFIGURATION_FILE, defaultOCIAuthConfFile);
+
+    public final static String COMPARTMENT_ID
+            = "org.jitsi.jigasi.transcription.oci.compartmentId";
+
+    private final String compartmentId;
+
+    public final static String DEFAULT_WEBSOCKET_URL = "ws://localhost:8000/ws";
+
+    private static BasicAuthenticationDetailsProvider authProvider;
+
+    static
+    {
+
+        try
+        {
+            authProvider = new ConfigFileAuthenticationDetailsProvider(ConfigFileReader.parse(configFilePath));
+        }
+        catch (IOException e)
+        {
+            logger.warn("Error while reading OCI configuration file, trying to use the instance's principal", e);
+        }
+
+        if (authProvider == null)
+        {
+            try
+            {
+                authProvider = new InstancePrincipalsAuthenticationDetailsProvider.
+                        InstancePrincipalsAuthenticationDetailsProviderBuilder().
+                        build();
+            }
+            catch (Exception e)
+            {
+                logger.error("Error while creating OCI instance principal provider", e);
+            }
+        }
+    }
+
+
+    /**
+     * The config value of the websocket to the speech-to-text service.
+     */
+    private final String websocketUrlConfig;
+
+
+    /**
+     * Create a TranscriptionService which will send audio to the OCI service
+     * platform to get a transcription
+     */
+    public OracleTranscriptionService()
+    {
+        websocketUrlConfig = JigasiBundleActivator.getConfigurationService()
+                .getString(WEBSOCKET_URL, DEFAULT_WEBSOCKET_URL);
+        compartmentId = JigasiBundleActivator.getConfigurationService()
+                .getString(COMPARTMENT_ID, "");
+    }
+
+    public boolean supportsLanguageRouting()
+    {
+        return false;
+    }
+
+
+    @Override
+    public boolean supportsStreamRecognition()
+    {
+        return true;
+    }
+
+    @Override
+    public boolean supportsFragmentTranscription()
+    {
+        return false;
+    }
+
+    @Override
+    public void sendSingleRequest(final TranscriptionRequest request,
+                                  final Consumer<TranscriptionResult> resultConsumer)
+    {
+        logger.warn("The OCI transcription service does not support single requests.");
+    }
+
+    @Override
+    public StreamingRecognitionSession initStreamingSession(
+            Participant participant)
+    {
+        return new OracleStreamingSession(websocketUrlConfig);
+    }
+
+    @Override
+    public boolean isConfiguredProperly()
+    {
+        return true;
+    }
+
+    public class OracleStreamingSession implements StreamingRecognitionSession, OracleRealtimeClientListener
+    {
+        private OracleRealtimeClient client;
+
+        private UUID uuid = UUID.randomUUID();
```

Comment:
This is the id for the transcriptions, right?
Can you switch to `Generators.timeBasedReorderedGenerator().generate();` so the transcriptions can be sorted by time as the rest of transcription providers?